sky/skylet/job_lib.py

```diff
@@ -147,7 +152,24 @@ def set_status(job_id: int, status: JobStatus) -> None:
     _CONN.commit()
 
 
-def get_status(job_id: int) -> Optional[JobStatus]:
+def set_status(job_id: int, status: JobStatus) -> None:
+    # TODO(mraheja): remove pylint disabling when filelock version updated
+    # pylint: disable=abstract-class-instantiated
+    with filelock.FileLock(_JOB_STATUS_LOCK.format(job_id)):
+        _set_status_no_lock(job_id, status)
+
+
+def set_job_started(job_id: int) -> None:
+    # TODO(mraheja): remove pylint disabling when filelock version updated.
+    # pylint: disable=abstract-class-instantiated
+    with filelock.FileLock(_JOB_STATUS_LOCK.format(job_id)):
+        _CURSOR.execute(
+            'UPDATE jobs SET status=(?), start_at=(?), end_at=NULL '
+            'WHERE job_id=(?)', (JobStatus.RUNNING.value, time.time(), job_id))
+        _CONN.commit()
+
+
+def get_status_no_lock(job_id: int) -> JobStatus:
```

Comment:
Add docstr saying: this may return a stale status if there's a concurrent update. Use get_status() to protect against that?

I still lean towards we should not expose this and used outside; but if we do, the docstring should give some guidance on when to call which func.